sdks/python/apache_beam/io/aws/clients/s3/boto3_client.py

```diff
@@ -0,0 +1,239 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from __future__ import absolute_import
+
+from apache_beam.io.aws.clients.s3 import messages
+
+try:
+  # pylint: disable=wrong-import-order, wrong-import-position
+  # pylint: disable=ungrouped-imports
+  import boto3
+
+except ImportError:
+  raise ImportError('Missing boto3 requirement')
```

Comment:
Sorry. I think another approach for this line could be:
```suggestion
  boto3 = None
```

Reply:
This file is the real, non-mocked boto3 client, so it only makes sense to use it with boto3 installed. If we catch the `ImportError` and silence it this way the user will just get a more cryptic error later (something like 'NoneType' object has no attribute 'client') which could be harder to debug. If it's our usage of a built-in error that's causing the problem, could we just change the type of error we're raising?

Reply:
We can also just ignore the original import error, that is just delete lines 27 and 28 entirely.

Reply:
You could add this, and in the client `__init__` method (where boto3 is actually called), add a line with something like:

```
assert boto3 is not None, 'Missing boto3 requirement'
```

This would prevent this import-time error, and let the tests pass, and give a reasonable error message during client intialization rather than at import. Thoughts?

Reply:
Works for me!